src/ast/function/all_operator.rs

```diff
@@ -0,0 +1,28 @@
+use super::Function;
+use crate::ast::Expression;
+
+/// A representation of the `AllOperator` operator in the database.
+#[derive(Debug, Clone, PartialEq)]
+pub struct AllOperator<'a> {
+    pub(crate) expr: Expression<'a>,
+}
+
+/// Wraps an expression into the ALL operator.
+///
+/// ```rust
+/// # use quaint::{ast::*, col, visitor::{Visitor, Postgres}};
+/// # fn main() -> Result<(), quaint::error::Error> {
+/// let query = Select::from_table("users").so_that(col!("name").equals(all_operator(col!("list"))));
+/// let (sql, _) = Postgres::build(query)?;
+/// assert_eq!(r#"SELECT "users".* FROM "users" WHERE "name" = ALL("list")"#, sql);
+/// # Ok(())
+/// # }
+/// ```
+pub fn all_operator<'a, T>(expr: T) -> Function<'a>
+where
+    T: Into<Expression<'a>>,
+{
+    let fun = AllOperator { expr: expr.into() };
+
+    fun.into()
+}
```

Comment:
So... `ALL` and `ANY` are apparently operators. I might sound dumb but this doesn't look like an operator to me... - `<left> operator <right>`
- `<left> operator`
- `operator <right>`

Those are all typical operator patterns as defined in most languages. But `ANY` and `ALL` have [this weird construct](https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-any/):

```
expresion operator ANY(subquery)
```

I'm not sure I get how `<left> <operator> <operator(ANY | ALL)> <right>` make any sense.

Anyway, because of that, I decided to treat `ANY` and `ALL` as functions, and not operator, which make it a lot easier to integrate with Quaint's assumptions of what an expression look like.

This is just to explain why I ended up with this very weird construct of `any_operator` and `all_operator` being functions 😭.

I'm open to suggestions, I know this is ugly

Reply:
Refactored ✅